src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/ArrayMarshaller.cs

```diff
@@ -58,7 +58,12 @@ public static ReadOnlySpan<T> GetManagedValuesSource(T[]? managed)
         /// <param name="numElements">The unmanaged element count.</param>
         /// <returns>The <see cref="Span{TUnmanagedElement}"/> of unmanaged elements.</returns>
         public static Span<TUnmanagedElement> GetUnmanagedValuesDestination(TUnmanagedElement* unmanaged, int numElements)
-            => new Span<TUnmanagedElement>(unmanaged, numElements);
```

Comment:
_Maybe_ it makes sense to allow the UnmanagedValuesDestination to have the length expected, but I lean towards making it empty for null.

Reply:
I'm not sure this is ever true. If the `unmanaged` is `null`, any non-zero value in `numElements` is a bug. What you have here is what I would expect from a defensive interop system. Users passing `(null, non_zero)` either have a bug or are trying to do something nefarious.

Reply:
This could be used in the reversed PInvoke case when marshalling elements of a `ref Span<T>` back to native. If there's a scenario like the motivating issue where the same length is used for two parameters and one is `null` we could end up with this returning `(null, non_zero)`.

Reply:
I misunderstood what you were trying to say here, I agree the null check should be here to be safe.